### ISSUE TITLE
关于仓库自带的Finetune代码target生成的疑惑

### ISSUE THREAD

**Issue by TankNee** [QwenLM/Qwen1.5#81]
https://github.com/QwenLM/Qwen1.5/blob/52cf91bd1c4eee35a039f3b50723b49cc431956e/examples/sft/finetune.py#L137-L164

根据我的理解，这里的target ids忽略的token仅仅是padding上去的token，并没有把prompt token和role token忽略掉，我用该脚本进行训练时loss高的离谱

**Comment by JustinLin610** [comment]
Base or chat model that you start with? Training without masking these elements will not lead to your loss problems. This is an example of the loss.

Base model
![image](https://github.com/QwenLM/Qwen1.5/assets/27664428/39101a84-78a1-4c70-b634-0885adc48d3b)

Chat model
![image](https://github.com/QwenLM/Qwen1.5/assets/27664428/53196693-f061-4566-be52-66e3dedc83ef)

Chat model will get higher loss cuz we trained the model those masking strategies and it takes some time to learn the role or prompt

**Comment by TankNee** [comment]
I'm very glad you can answer my question.

In my experiment, I use Qwen-14B-Chat model and my finetune method is LoRA.

Here's how the loss changes when I don't add the ignore tag
![image](https://github.com/QwenLM/Qwen1.5/assets/43956168/2dd84860-f6f1-4a70-b313-5fee36614afe)

When I changed the logic of the target generation part to look like this, the loss dropped much faster.

![CleanShot 2024-02-25 at 17 10 53@2x](https://github.com/QwenLM/Qwen1.5/assets/43956168/fa87595c-361d-4895-9ed2-2b92d038a26c)

But I still have a doubt. The last message in the input messages is a label. According to the original logic, Except for the difference between padding and ignore idx, the target and input id are exactly the same?
